### DISCUSSION TITLE
Migration of graphQL java from 11.0 to 16.1

### DISCUSSION THREAD

**Discussion by vyj7** [graphql-java/graphql-java#2199] [Q&A]
Hi, I am currently upgrading Java-graphql version libs. I am facing an error during the application start.

I am upgrading the following :

 <graphql-java.version>11.0</graphql-java.version>
 <graphql-java-tools.version>5.2.4</graphql-java-tools.version>
 <graphql-java-servlet.version>6.1.3</graphql-java-servlet.version>

to 

 <graphql-java.version>16.1</graphql-java.version>
 <graphql-java-tools.version>5.2.4</graphql-java-tools.version>
 <graphql-java-servlet.version>11.0.0</graphql-java-servlet.version>

I am getting these 2 errors:

ANTLR Tool version 4.8 used for code generation does not match the current runtime version 4.7.2ANTLR Runtime version 4.8 used for parser compilation does not match the current runtime version 4.7.2ANTLR Tool version 4.8 used for code generation does not match the current runtime version 4.7.2ANTLR Runtime version 4.8 used for parser compilation does not match the current runtime version 4.7.211:59:14.692 |  WARN | o.s.b.w.s.c.AnnotationConfigServletWebServerApplicationContext | Exception encountered during context initialization - cancelling refresh attempt: org.springframework.context.ApplicationContextException: Unable to start web server; nested exception is java.lang.RuntimeException: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'graphQLServletRegistrationBean' defined in class path resource [com/oembedler/moon/graphql/boot/GraphQLWebAutoConfiguration.class]: Unsatisfied dependency expressed through method 'graphQLServletRegistrationBean' parameter 0; 

***************************
APPLICATION FAILED TO START
***************************

Description:

An attempt was made to call a method that does not exist. The attempt was made from the following location:

    com.coxautodev.graphql.tools.RootTypeInfo$Companion.fromSchemaDefinitions(RootTypeInfo.kt:16)

The following method did not exist:

    graphql.language.OperationTypeDefinition.getType()Lgraphql/language/Type;

The method's class, graphql.language.OperationTypeDefinition, is available from the following locations:

    jar:file:/Users/vayuj/.m2/repository/com/graphql-java/graphql-java/16.1/graphql-java-16.1.jar!/graphql/language/OperationTypeDefinition.class

It was loaded from the following location:

    file:/Users/vayuj/.m2/repository/com/graphql-java/graphql-java/16.1/graphql-java-16.1.jar

Action:

Correct the classpath of your application so that it contains a single, compatible version of graphql.language.OperationTypeDefinition

**Comment by andimarek** [comment]
HI @vyj7,

this project is only for graphql java itself and your error comes from `com.coxautodev.graphql.tools.RootTypeInfo$Companion.fromSchemaDefinitions(RootTypeInfo.kt:16)` which is not GraphQL Java.

The Kickstarter project is probably what you are looking for: https://www.graphql-java-kickstart.com/

**Reply by vyj7** [reply]
@andimarek  Thank you . Will check if I can find anything there. Detailed Stack Trace points to Java Graphql Tools .
Also I saw the change in 11.0 and 16.1.  
This class 
`public class OperationTypeDefinition extends AbstractNode<OperationTypeDefinition> implements NamedNode<OperationTypeDefinition> {
  private final String name;
  private final TypeName typeName;
  public static final String CHILD_TYPE_NAME = "typeName";
`

was different in 11.0. Here Tyname has been introduce 

`Unsatisfied dependency expressed through constructor parameter 0; nested exception is org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'graphQLSchema' defined in class path resource [com/oembedler/moon/graphql/boot/GraphQLJavaToolsAutoConfiguration.class]: Unsatisfied dependency expressed through method 'graphQLSchema' parameter 0; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'schemaParser' defined in class path resource [com/oembedler/moon/graphql/boot/GraphQLJavaToolsAutoConfiguration.class]: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [com.coxautodev.graphql.tools.SchemaParser]: Factory method 'schemaParser' threw exception; nested exception is java.lang.NoSuchMethodError: graphql.language.OperationTypeDefinition.getType()Lgraphql/language/Type;
`
